compiler-rt/lib/builtins/fp_extend.h

```diff
@@ -39,6 +43,16 @@ static __inline int src_rep_t_clz(src_rep_t a) {
 #endif
 }
 
+#elif defined SRC_80
+typedef long double src_t;
+typedef __uint128_t src_rep_t;
+#define SRC_REP_C (__uint128_t)
+// sign bit, exponent and significand occupy the lower 80 bits.
+static const int srcBits = 80;
+// significand stores the integer bit.
+static const int srcSigBits = 64;
+static const int srcSigFracBits = 63;
```

Comment:
To be specific, I think the approach here is making the patch much harder to understand than is really necessary. 
 I think defining two different constants representing the size of the significand, and integrating the logic to handle these directly into the routines, is confusing.

I'd like to see a helper function "extract significant fraction bits" or something like that.  And a function to construct a float given the exponent, and significant fraction bits.  Then the rest of the code can use those helpers, instead of making it directly aware of the extra bit in fp80 representations.

Reply:
ok, I initially started with something similar to what you are suggesting,
but this turned out to be quite far from what the existing template.
However, I agree, this probably would be cleaner, I'll try to udpate the PR ~soon.

Reply:
@efriedma-quic - I'm working on giving it another try, but want to point out - that this would be essentially a rewrite of the existing template (so the changes will be quite invasive). Unfortunately, the current implementation of __extendXfYf2__ (even without any of my changes) is not that simple and doesn't use the model that you'd like to see.

Reply:
I don't see an issue with making relatively extensive changes here; the old code is pretty fragile and hard to read anyway.  (I mean, we say "invasive", but it's not affecting anything outside the implementation itself.)

Updated extend looks good; it's shorter and easier to read, even with the new functionality.

Reply:
Updated "truncate" as well.